### PR TITLE
Add functionality requested for Shattered Paradise

### LINKED ISSUE #10978: Make RejectsOrders upgradable
Requested by @ABrandau.

### PR DESCRIPTION
**Pull request by obrakmann**
Closes #10978.

//cc @ABrandau.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/GivesBounty.cs [hunk 1]
```diff
@@ -9,29 +9,48 @@
...
+
+		public GivesBounty(Actor self, GivesBountyInfo info)
+		{
+			this.info = info;
+		}
+
+		int GetMultiplier(Actor self)
 		{
 			// returns 100's as 1, so as to keep accuracy for longer.
-			var info = self.Info.TraitInfo<GivesBountyInfo>();
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/GivesBounty.cs]: :+1: Well spotted.

**Comment by Mailaender** [comment]
:+1:

**Comment by Nelax** [comment]
:+1: Works as intended. Mind you the testcase doesn't work very well as a good example.

File: mods/ra/rules/aircraft.yaml [hunk 2]
```diff
@@ -255,6 +255,10 @@ HELI:
 		Description: Helicopter gunship armed\nwith multi-purpose missiles.\n  Strong vs Tanks, Aircraft\n  Weak vs Infantry
 	Health:
 		HP: 120
+	RejectsOrders:
+		Except: Attack
```
Comment (Phrohdoh) [inline mods/ra/rules/aircraft.yaml]: This isn't working for me.
Perhaps I am missing something.

I am able to give non-Attack orders to my longbow just fine (as long as it is airborne).

Comment (obrakmann) [inline mods/ra/rules/aircraft.yaml]: Yes. `RejectsOrders` gets disabled with the airborne upgrade, hence it doesn't reject orders when airborne.

You should only be able to issue attack orders while the heli is on the ground, not any move or other orders.

**Comment by obrakmann** [comment]
Removed the test case (was commit ae403c8, if anyone wants to take a look again).

**Comment by GraionDilach** [comment]
:+1:

**Comment by reaperrr** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/7eb9f03c390e3c3843e8de9618c8282a46363545...920d054fe55ffac01a89e08aab7b4aef8c4c692d)

**Pull request merged**